test/generate-xml.sh

```diff
@@ -43,32 +43,40 @@ if [ ! -x $mbus_parse_hex ]; then
     exit 3
 fi
 
-for hexfile in "$directory"/*.hex;  do
-    if [ ! -f "$hexfile" ]; then
-        continue
-    fi
+generate_xml() {
+    directory="$1"
+    hexfile="$2"
+    mode="$3"
 
     filename=`basename $hexfile .hex`
 
+    if [ $mode = "normalized" ]; then
+        options="-n"
+        suffix=".norm.xml"
+    else
+        options=""
+        suffix=".xml"
+    fi
+
     # Parse hex file and write XML in file
-    $mbus_parse_hex "$hexfile" > "$directory/$filename.xml.new"
+    $mbus_parse_hex $options "$hexfile" > "$directory/$filename$suffix.new"
```

Comment:
Personal opinion: I would have a dot after filename, and probably keep "xml.new" down here.
Since that is invariant (does not differ over the two cases.
 
```
 if [ $mode = "normalized" ]; then
        options="-n"
        suffix="norm"
    else
        options=""
        suffix=""
    fi
  ...
    $mbus_parse_hex $options "$hexfile" > "$directory/$filename.${suffix}xml.new"
```

Reply:
I didn't understand the keep xml.new thing. In case the output differ we keep the new file. So we need the suffix, otherwise we will overwrite the new default output.

Reply:
What I was trying to say, is that both modes "default" and "normalized" will name the files withe the ending ".xml.new", 



 I would write it like this:

```
options = ""
if [ "$mode" = "default" ]; then
    mode=""
elif [ "$mode" = "normalized" ];
    mode=".norm"
    options="-n"
else
    echo "ERROR: Unknown mode $mode."
    return 1
fi
...
$mbus_parse_hex $options "$hexfile" > "$directory/$filename$mode.xml.new"

diff -u "$directory/$filename$mode.xml" "$directory/$filename$mode.xm.new" 2> /dev/null > "$directory/$filename$mode.dif"

```